.github/workflows/check.yml

```diff
@@ -56,7 +56,23 @@ jobs:
 
       - name: Run Tests
         run: bundle exec fastlane ${{ matrix.lane }}
+
+      - name: Check Static Analyzer Output
+        run: |
+          if [[ -z $(find ./derived_data -name "report-*.html") ]]; then
+            echo "Static Analyzer found no issues."
+          else
+            echo "Static Analyzer found some issues. Failing build."
+            exit 1
+          fi
       
+      - name: Static Analyzer Reports Artifact
+        if: failure()
```

Comment:
This is, unfortunately, a slightly flawed approach as this step will execute no matter which of the steps above it failed. So, for example, in [this run](https://github.com/ably/ably-cocoa/runs/4275016222?check_suite_focus=true):

```
Run actions/upload-artifact@v2
  with:
    name: static-analyzer-reports-test_macOS
    path: ./derived_data/**/StaticAnalyzer/**/*.html
    if-no-files-found: warn
  env:
    LC_CTYPE: en_US.UTF-8
    LANG: en_US.UTF-8
    ABLY_ENV: sandbox
Warning: No files were found with the provided path: ./derived_data/**/StaticAnalyzer/**/*.html. No artifacts will be uploaded.
```

Where the build failed at "Run Tests", not "Check Static Analyzer Output".

Reply:
https://github.com/ably/ably-cocoa/pull/1213/commits/16dd4de638cf99b45810d9d33d45f2d426e575ff

Reply:
I'm not really sure how https://github.com/ably/ably-cocoa/commit/16dd4de638cf99b45810d9d33d45f2d426e575ff solves this, as the later step guarded by `if: always()` will still execute - which was what I was commented on. 😕 

Have I missed something?

The `exit 1` in your new step will fail that step, failing the job, but then the job was already failed.

Reply:
>later step guarded by if: always() will still execute - which was what I was commented on
I believe your initial comment was on line 70, which is "Static Analyzer Reports Uploading" step now, but "later step" you're referring is "Xcodebuild Logs Artifact" which is different thing (uploads all build logs, not SA html reports).

Now how it works: I fail build if reports found in the "Check Static Analyzer Output" step. Then, in case of failure next step "Static Analyzer Reports Uploading" is activated. And then, because it's impossible to use `run` section with pre-build step `uses` section, I've added one more step only to spit the message about failure and exit with... failure. Now I see that it is a complete nonsense, and it seems I've just failed to fail build failure here 🤷‍♂️😆 The real fix was in replacing `if: failure()` with `if: steps.analyzer-output.outcome == 'failure'` in "Static Analyzer Reports Artifact" (renamed to "Static Analyzer Reports Uploading").

Reply:
@QuintinWillison 122720c2ee2c1b0030cd00b6dc8fa12025f627dd